lib/utils/ssh.ts

```diff
@@ -147,9 +147,22 @@ export async function spawnSshAndThrowOnError(
 		// to return exit code 1. In these cases, print a short one-line warning
 		// message, and exits the CLI process with the same error code.
 		process.exitCode = exitCode;
-		const msg = exitSignal
-			? `Warning: ssh process was terminated with signal "${exitSignal}"`
-			: `Warning: ssh process exited with non-zero code "${exitCode}"`;
-		throw new ExpectedError(msg);
+		throw new ExpectedError(sshErrorMessage(exitSignal!, exitCode!));
```

Comment:
hmm Now my focus in on those two exclamation marks. They tell the tsc compiler "I know that this value will not be undefined", but in fact, for example `exitSignal` will most of the time actually be undefined, and this could mask errors in the implementation of `sshErrorMessage`. I suggest replacing the exclamation marks here with question marks in the signature of `sshErrorMessage`:
```
    throw new ExpectedError(sshErrorMessage(exitSignal, exitCode));
...

function sshErrorMessage(exitSignal?: string, exitCode?: number) {
...
```
Apparently this would not require any changes in the body of `sshErrorMessage`.

Reply:
I should've looked into ! further (New to typescript). The ? looks appropriate as undefined will simply pass through the switch statement.